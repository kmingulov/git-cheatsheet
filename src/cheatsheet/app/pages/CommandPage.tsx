import React, { PureComponent, ReactElement } from 'react';
import { Button, FormControl, InputGroup } from 'react-bootstrap';
import { MdClear, MdSearch } from 'react-icons/md';

import commands from 'cheatsheet/command-data/commands';
import { CommandStore, LunrCommandStore } from 'cheatsheet/command-store';
import { CommandTable } from 'cheatsheet/command-table';
import { staticComponent } from 'cheatsheet/util/componentUtils';

const StaticInputGroup = staticComponent(InputGroup);

type State = Readonly<{ searchTerm: string }>;
export class CommandPage extends PureComponent<object, State> {
  public readonly state: State = { searchTerm: '' };

  private store: CommandStore = new LunrCommandStore(commands);

  public render(): ReactElement {
    return (
      <div>
        <StaticInputGroup className='searchBox'>
          <InputGroup.Prepend>
            <InputGroup.Text>
              <MdSearch/>
            </InputGroup.Text>
          </InputGroup.Prepend>
          <FormControl
            placeholder='Searchâ€¦'
            onChange={ this.onChange }
          />
          <InputGroup.Append>
            <Button
              variant='secondary'
              onClick={ this.onClear }
            >
              <MdClear/>
            </Button>
          </InputGroup.Append>
        </StaticInputGroup>

        <CommandTable store={ this.store } searchTerm={ this.state.searchTerm }/>
      </div>
    );
  }

  private onChange = (event: any): void => {
    this.setState({ searchTerm: event.target.value });
  }

  private onClear = (): void => {
    this.setState({ searchTerm: '' });
  }
}
