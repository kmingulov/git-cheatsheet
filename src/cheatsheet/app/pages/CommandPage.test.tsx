import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';

import { CommandPage } from './CommandPage';

jest.mock('react-icons/md', () => ({
  MdClear: () => <div/>,
  MdSearch: () => <div/>,
}));
jest.mock('cheatsheet/command-store');
jest.mock('cheatsheet/command-table');

Enzyme.configure({adapter: new Adapter()});

describe(CommandPage, () => {
  it('renders correctly', () => {
    const component = mount(<CommandPage/>);
    expect(component).toMatchSnapshot();
  });

  it('allows searching', () => {
    const component = mount(<CommandPage/>);
    expect(component).toMatchSnapshot();

    component.find('FormControl').simulate('change', {
      target: {value: 'search term'},
    });
    expect(component).toMatchSnapshot();

    component.find('Button').simulate('click');
    expect(component).toMatchSnapshot();
  });
});
