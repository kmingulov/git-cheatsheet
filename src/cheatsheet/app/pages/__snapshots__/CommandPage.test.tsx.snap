// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommandPage allows searching 1`] = `
<CommandPage>
  <div>
    <ForwardRef(Bootstrap(InputGroup))
      className="searchBox"
    >
      <InputGroup
        as="div"
        bsPrefix="input-group"
        className="searchBox"
      >
        <div
          className="searchBox input-group"
        >
          <staticComponent()>
            <ForwardRef(Bootstrap(InputGroupPrepend))>
              <InputGroupPrepend
                bsPrefix="input-group-prepend"
              >
                <div
                  className="input-group-prepend"
                >
                  <ForwardRef(Bootstrap(InputGroupText))>
                    <InputGroupText
                      bsPrefix="input-group-text"
                    >
                      <span
                        className="input-group-text"
                      >
                        <MdSearch>
                          <div />
                        </MdSearch>
                      </span>
                    </InputGroupText>
                  </ForwardRef(Bootstrap(InputGroupText))>
                </div>
              </InputGroupPrepend>
            </ForwardRef(Bootstrap(InputGroupPrepend))>
          </staticComponent()>
          <ForwardRef(ContextTransform)
            onChange={[Function]}
            placeholder="Search…"
            value=""
          >
            <ForwardRef(Bootstrap(FormControl))
              onChange={[Function]}
              placeholder="Search…"
              value=""
            >
              <FormControl
                as="input"
                bsPrefix="form-control"
                inputRef={null}
                onChange={[Function]}
                placeholder="Search…"
                value=""
              >
                <input
                  className="form-control"
                  onChange={[Function]}
                  placeholder="Search…"
                  value=""
                />
              </FormControl>
            </ForwardRef(Bootstrap(FormControl))>
          </ForwardRef(ContextTransform)>
          <staticComponent()>
            <ForwardRef(Bootstrap(InputGroupAppend))>
              <InputGroupAppend
                bsPrefix="input-group-append"
              >
                <div
                  className="input-group-append"
                >
                  <ForwardRef(Bootstrap(Button))
                    onClick={[Function]}
                    variant="secondary"
                  >
                    <Button
                      active={false}
                      bsPrefix="btn"
                      disabled={false}
                      innerRef={null}
                      onClick={[Function]}
                      type="button"
                      variant="secondary"
                    >
                      <button
                        className="btn btn-secondary"
                        disabled={false}
                        onClick={[Function]}
                        type="button"
                      >
                        <MdClear>
                          <div />
                        </MdClear>
                      </button>
                    </Button>
                  </ForwardRef(Bootstrap(Button))>
                </div>
              </InputGroupAppend>
            </ForwardRef(Bootstrap(InputGroupAppend))>
          </staticComponent()>
        </div>
      </InputGroup>
    </ForwardRef(Bootstrap(InputGroup))>
    <CommandTable
      searchTerm=""
      store={
        LunrCommandStore {
          "getAll": [MockFunction],
          "search": [MockFunction],
        }
      }
    />
  </div>
</CommandPage>
`;

exports[`CommandPage allows searching 2`] = `
<CommandPage>
  <div>
    <ForwardRef(Bootstrap(InputGroup))
      className="searchBox"
    >
      <InputGroup
        as="div"
        bsPrefix="input-group"
        className="searchBox"
      >
        <div
          className="searchBox input-group"
        >
          <staticComponent()>
            <ForwardRef(Bootstrap(InputGroupPrepend))>
              <InputGroupPrepend
                bsPrefix="input-group-prepend"
              >
                <div
                  className="input-group-prepend"
                >
                  <ForwardRef(Bootstrap(InputGroupText))>
                    <InputGroupText
                      bsPrefix="input-group-text"
                    >
                      <span
                        className="input-group-text"
                      >
                        <MdSearch>
                          <div />
                        </MdSearch>
                      </span>
                    </InputGroupText>
                  </ForwardRef(Bootstrap(InputGroupText))>
                </div>
              </InputGroupPrepend>
            </ForwardRef(Bootstrap(InputGroupPrepend))>
          </staticComponent()>
          <ForwardRef(ContextTransform)
            onChange={[Function]}
            placeholder="Search…"
            value="search term"
          >
            <ForwardRef(Bootstrap(FormControl))
              onChange={[Function]}
              placeholder="Search…"
              value="search term"
            >
              <FormControl
                as="input"
                bsPrefix="form-control"
                inputRef={null}
                onChange={[Function]}
                placeholder="Search…"
                value="search term"
              >
                <input
                  className="form-control"
                  onChange={[Function]}
                  placeholder="Search…"
                  value="search term"
                />
              </FormControl>
            </ForwardRef(Bootstrap(FormControl))>
          </ForwardRef(ContextTransform)>
          <staticComponent()>
            <ForwardRef(Bootstrap(InputGroupAppend))>
              <InputGroupAppend
                bsPrefix="input-group-append"
              >
                <div
                  className="input-group-append"
                >
                  <ForwardRef(Bootstrap(Button))
                    onClick={[Function]}
                    variant="secondary"
                  >
                    <Button
                      active={false}
                      bsPrefix="btn"
                      disabled={false}
                      innerRef={null}
                      onClick={[Function]}
                      type="button"
                      variant="secondary"
                    >
                      <button
                        className="btn btn-secondary"
                        disabled={false}
                        onClick={[Function]}
                        type="button"
                      >
                        <MdClear>
                          <div />
                        </MdClear>
                      </button>
                    </Button>
                  </ForwardRef(Bootstrap(Button))>
                </div>
              </InputGroupAppend>
            </ForwardRef(Bootstrap(InputGroupAppend))>
          </staticComponent()>
        </div>
      </InputGroup>
    </ForwardRef(Bootstrap(InputGroup))>
    <CommandTable
      searchTerm="search term"
      store={
        LunrCommandStore {
          "getAll": [MockFunction],
          "search": [MockFunction],
        }
      }
    />
  </div>
</CommandPage>
`;

exports[`CommandPage allows searching 3`] = `
<CommandPage>
  <div>
    <ForwardRef(Bootstrap(InputGroup))
      className="searchBox"
    >
      <InputGroup
        as="div"
        bsPrefix="input-group"
        className="searchBox"
      >
        <div
          className="searchBox input-group"
        >
          <staticComponent()>
            <ForwardRef(Bootstrap(InputGroupPrepend))>
              <InputGroupPrepend
                bsPrefix="input-group-prepend"
              >
                <div
                  className="input-group-prepend"
                >
                  <ForwardRef(Bootstrap(InputGroupText))>
                    <InputGroupText
                      bsPrefix="input-group-text"
                    >
                      <span
                        className="input-group-text"
                      >
                        <MdSearch>
                          <div />
                        </MdSearch>
                      </span>
                    </InputGroupText>
                  </ForwardRef(Bootstrap(InputGroupText))>
                </div>
              </InputGroupPrepend>
            </ForwardRef(Bootstrap(InputGroupPrepend))>
          </staticComponent()>
          <ForwardRef(ContextTransform)
            onChange={[Function]}
            placeholder="Search…"
            value=""
          >
            <ForwardRef(Bootstrap(FormControl))
              onChange={[Function]}
              placeholder="Search…"
              value=""
            >
              <FormControl
                as="input"
                bsPrefix="form-control"
                inputRef={null}
                onChange={[Function]}
                placeholder="Search…"
                value=""
              >
                <input
                  className="form-control"
                  onChange={[Function]}
                  placeholder="Search…"
                  value=""
                />
              </FormControl>
            </ForwardRef(Bootstrap(FormControl))>
          </ForwardRef(ContextTransform)>
          <staticComponent()>
            <ForwardRef(Bootstrap(InputGroupAppend))>
              <InputGroupAppend
                bsPrefix="input-group-append"
              >
                <div
                  className="input-group-append"
                >
                  <ForwardRef(Bootstrap(Button))
                    onClick={[Function]}
                    variant="secondary"
                  >
                    <Button
                      active={false}
                      bsPrefix="btn"
                      disabled={false}
                      innerRef={null}
                      onClick={[Function]}
                      type="button"
                      variant="secondary"
                    >
                      <button
                        className="btn btn-secondary"
                        disabled={false}
                        onClick={[Function]}
                        type="button"
                      >
                        <MdClear>
                          <div />
                        </MdClear>
                      </button>
                    </Button>
                  </ForwardRef(Bootstrap(Button))>
                </div>
              </InputGroupAppend>
            </ForwardRef(Bootstrap(InputGroupAppend))>
          </staticComponent()>
        </div>
      </InputGroup>
    </ForwardRef(Bootstrap(InputGroup))>
    <CommandTable
      searchTerm=""
      store={
        LunrCommandStore {
          "getAll": [MockFunction],
          "search": [MockFunction],
        }
      }
    />
  </div>
</CommandPage>
`;

exports[`CommandPage renders correctly 1`] = `
<CommandPage>
  <div>
    <ForwardRef(Bootstrap(InputGroup))
      className="searchBox"
    >
      <InputGroup
        as="div"
        bsPrefix="input-group"
        className="searchBox"
      >
        <div
          className="searchBox input-group"
        >
          <staticComponent()>
            <ForwardRef(Bootstrap(InputGroupPrepend))>
              <InputGroupPrepend
                bsPrefix="input-group-prepend"
              >
                <div
                  className="input-group-prepend"
                >
                  <ForwardRef(Bootstrap(InputGroupText))>
                    <InputGroupText
                      bsPrefix="input-group-text"
                    >
                      <span
                        className="input-group-text"
                      >
                        <MdSearch>
                          <div />
                        </MdSearch>
                      </span>
                    </InputGroupText>
                  </ForwardRef(Bootstrap(InputGroupText))>
                </div>
              </InputGroupPrepend>
            </ForwardRef(Bootstrap(InputGroupPrepend))>
          </staticComponent()>
          <ForwardRef(ContextTransform)
            onChange={[Function]}
            placeholder="Search…"
            value=""
          >
            <ForwardRef(Bootstrap(FormControl))
              onChange={[Function]}
              placeholder="Search…"
              value=""
            >
              <FormControl
                as="input"
                bsPrefix="form-control"
                inputRef={null}
                onChange={[Function]}
                placeholder="Search…"
                value=""
              >
                <input
                  className="form-control"
                  onChange={[Function]}
                  placeholder="Search…"
                  value=""
                />
              </FormControl>
            </ForwardRef(Bootstrap(FormControl))>
          </ForwardRef(ContextTransform)>
          <staticComponent()>
            <ForwardRef(Bootstrap(InputGroupAppend))>
              <InputGroupAppend
                bsPrefix="input-group-append"
              >
                <div
                  className="input-group-append"
                >
                  <ForwardRef(Bootstrap(Button))
                    onClick={[Function]}
                    variant="secondary"
                  >
                    <Button
                      active={false}
                      bsPrefix="btn"
                      disabled={false}
                      innerRef={null}
                      onClick={[Function]}
                      type="button"
                      variant="secondary"
                    >
                      <button
                        className="btn btn-secondary"
                        disabled={false}
                        onClick={[Function]}
                        type="button"
                      >
                        <MdClear>
                          <div />
                        </MdClear>
                      </button>
                    </Button>
                  </ForwardRef(Bootstrap(Button))>
                </div>
              </InputGroupAppend>
            </ForwardRef(Bootstrap(InputGroupAppend))>
          </staticComponent()>
        </div>
      </InputGroup>
    </ForwardRef(Bootstrap(InputGroup))>
    <CommandTable
      searchTerm=""
      store={
        LunrCommandStore {
          "getAll": [MockFunction],
          "search": [MockFunction],
        }
      }
    />
  </div>
</CommandPage>
`;
